File: bookstore/app/models/book.js
import { tracked } from '@glimmer/tracking';

export default class Book {
    @tracked id;
    @tracked title;
    @tracked isbn;
    @tracked publishedDate;
    @tracked authorList;

    constructor(o = {
        id: 0,
        title: '',
        isbn: '',
        publishedDate: new Date(),
        authorList: [{
            firstName: '',
            lastName: '',
            doB: ''
        }
        ]
    }) {
        this.id = o.id;
        this.title = o.title;
        this.isbn = o.isbn;
        this.publishedDate = o.publishedDate;
        this.authorList = o.authorList;
    }

}

File: Proof-Of-Concept/src/main/java/com/craftable/ProofOfConcept/Model/Book.java
package com.craftable.ProofOfConcept.Model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY )
    private int id;
    private String title;
    @OneToMany(cascade = CascadeType.ALL )
    @JoinColumn(name = "bookId")
    private List<Author> authorList;
    private String ISBN;
    private Date publishedDate;


}
